Google.PowerShell/Storage/GcsObject.cs

```diff
@@ -866,24 +889,26 @@ private class ParameterSetNames
         /// Name of the object to write to.
         /// </para>
         /// </summary>
-        [Parameter(Position = 1, Mandatory = true)]
+        [Parameter(ParameterSetName = ParameterSetNames.ByNameFromString, Position = 1, Mandatory = true)]
+        [Parameter(ParameterSetName = ParameterSetNames.ByNameFromFile, Position = 1, Mandatory = true)]
         public string ObjectName { get; set; }
 
         /// <summary>
         /// <para type="description">
         /// Text content to write to the Storage object. Ignored if File is specified.
         /// </para>
         /// </summary>
-        [Parameter(ParameterSetName = ParameterSetNames.FromString,
-            Position = 2, ValueFromPipeline = true)]
-        public string Contents { get; set; }
+        [Parameter(ParameterSetName = ParameterSetNames.ByNameFromString, ValueFromPipeline = true, Mandatory = false)]
+        [Parameter(ParameterSetName = ParameterSetNames.ByObjectFromString, ValueFromPipeline = true, Mandatory = false)]
```

Comment:
I was under the impression that only a single parameter per parameter set could be `ValueFromPipeline = true`. Does the fact that they are different types make this ok?